### PR TITLE
Fix edge cases of link validation and conversion

### LINKED ISSUE #1820: Emails are wrongly validated as links
**Tell us about your environment**
MacOS Monterey 12.0.1
* **MarkBind Version:** v3.1.1

**What did you do? Please include the actual source code causing the issue.**
Place emails in any markdown file, e.g.:
```html
email@gmail.com
student@u.nus.edu
with.dot@mail.sg
```

**What did you expect to happen?**
The emails should not be logged as warnings for invalid intralinks.

**What actually happened? Please include the actual, raw output.**
Screenshot 2022-03-11 at 8 25 33 PM

The email is prepended with a filepath (e.g. `userGuide/mailto:`), which is confusing since this prefix does not appear in the source files at all, and doing a search on `userGuide/mailto:email@gmail.com` will not give any results for an author trying to find where the invalid link is.

This is because `convertRelativeLinks` is called on all links, including emails, so a filepath is prepended onto emails. 
I think a possible fix would be to check for emails (conveniently, they can be identified by a `mailto:` prefix) and not convert them to relative links, and possibly exclude them from `validateIntralink` calls.

### LINKED ISSUE #1874: Broken links reported when serving MarkBind docs
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Any related issues?

#1820 for email

### Tell us about your environment

Windows 10

### MarkBind version

3.1.1

### What did you do? Describe the bug

Serve the MarkBind docs locally, and saw 3 broken links reported.

### Steps to reproduce

checkout to the master branch of the markbind repo
cd docs
markbind serve

### Expected behavior

No broken links

### Actual behavior

image

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by tlylt**
**What is the purpose of this pull request?**

- [ ] Documentation update
- [x] Bug fix
- [ ] Feature addition or enhancement
- [ ] Code maintenance
- [ ] Others, please explain:

**Overview of changes:**
Fixes #1820
Fixes #1874

- Exclude Email/Tel links from Intralink validation & relative link conversion (need not append base URL)
- Fix validation for file path with space, which is treated as `%20`
- Fix an actual invalid intralink in our dev docs
- Update tests

**Anything you'd like to highlight / discuss:**
nil
**Testing instructions:**
Unit test added in `packages/core`
- run `npm run test` in that directory

**Proposed commit message: (wrap lines at 72 characters)**
Fix edge cases of link validation and conversion

Invalid intralinks have been reported in MarkBind docs. The
reasons are 1. actual invalid link 2. file path contains a space
3. Email link wrongly validated.

Let's fix all three link validation & conversion errors.
File paths with `%20` will be correctly checked for
existence. Email(tel) links will no longer be appended
with the base URL or validated as intralink.

The removal of invalid link behavior will allow
the link & link validation mechanism to function
properly.

---

**Checklist:** :ballot_box_with_check:

- [ ] Updated the documentation for feature additions and enhancements
- [x] Added tests for bug fixes or features
- [x] Linked all related issues
- [ ] No unrelated changes

### CODE REVIEW

File: packages/core/src/html/linkProcessor.js [hunk 1]
```diff
@@ -148,7 +154,10 @@ function isValidFileAsset(resourcePath, config) {
  * @returns {string} these string return values are for unit testing purposes only
  */
 function validateIntraLink(resourcePath, cwf, config) {
-  if (!resourcePath || urlUtil.isUrl(resourcePath) || resourcePath.startsWith('#')) {
+  if (!resourcePath
+    || urlUtil.isUrl(resourcePath)
+    || resourcePath.startsWith('#')
+    || /^(?:mailto:|tel:)/i.test(resourcePath)) {
```
Comment (jovyntls) [inline packages/core/src/html/linkProcessor.js:160]: Another possible change would be extracting all the `/^(?:mailto:|tel:)/i.test(resourcePath)` to a helper function in linkProcessor, though I'm not sure if that would be overcomplicating. Personally I think either way is fine!

**Review by jovyntls — COMMENTED** [review]
Looks good @tlylt , thanks for the fixes! Minor comments below

**Comment by kaixin-hc** [comment]
LGTM! Particularly timely fix since improving broken link reporting was one of the agenda items for V4 (#1418) - though this isn't one of the issues addressed there it's still related.

**Review by kaixin-hc — APPROVED** [review]

**Review by jovyntls — APPROVED** [review]
LGTM! The `isIntraLink` function is a great addition :)
